### PR TITLE
dep injection helloworld

### PR DESCRIPTION
**Pull request by frrist**
This fails to work and I'm not sure why. Steps to reproduce failure:
`lotus daemon`
< new terminal >
`lotus helloworld hello`
```
2021-02-23T16:59:02.093-0800    ERROR   rpc     go-jsonrpc@v0.1.4-0.20210217175800-45ea43ac2bec/handler.go:158      panic in rpc method 'Filecoin.Hello': runtime error: invalid memory address or nil pointer dereference
github.com/filecoin-project/go-jsonrpc.doCall.func1
        /home/frrist/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.1.4-0.20210217175800-45ea43ac2bec/handler.go:158
runtime.gopanic
        /home/frrist/.apps/go/src/runtime/panic.go:969
runtime.panicmem
        /home/frrist/.apps/go/src/runtime/panic.go:212
runtime.sigpanic
        /home/frrist/.apps/go/src/runtime/signal_unix.go:742
github.com/filecoin-project/lotus/api/apistruct.(*HelloWorldStruct).Hello
        /home/frrist/src/github.com/filecoin-project/lotus/api/apistruct/struct.go:483
reflect.Value.call
        /home/frrist/.apps/go/src/reflect/value.go:476
reflect.Value.Call
        /home/frrist/.apps/go/src/reflect/value.go:337
github.com/filecoin-project/go-jsonrpc.doCall
        /home/frrist/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.1.4-0.20210217175800-45ea43ac2bec/handler.go:162
github.com/filecoin-project/go-jsonrpc.(*RPCServer).handle
        /home/frrist/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.1.4-0.20210217175800-45ea43ac2bec/handler.go:259
2021-02-23T16:59:02.093-0800    ERROR   rpc     go-jsonrpc@v0.1.4-0.20210217175800-45ea43ac2bec/server.go:95RPC Error: fatal error calling 'Filecoin.Hello': panic in rpc method 'Filecoin.Hello': runtime error: invalid memory address or nil pointer dereference

```

### CODE REVIEW

**Comment by Kubuxu** [comment]
Maybe try adding it to: `api/apistruct/permissioned.go`

File: metrics/proxy.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ func MetricedFullAPI(a api.FullNode) api.FullNode {
 	var out apistruct.FullNodeStruct
 	proxy(a, &out.Internal)
 	proxy(a, &out.CommonStruct.Internal)
+	//proxy(a, &out.HelloWorldStruct.Internal)
```
Comment (frrist) [inline metrics/proxy.go:24]: This line is required in order for the hello world dependency to work. Without it, the client will panic when calling the `Hello` method. Smells a bit funky :cheese:

Comment (frrist) [reply metrics/proxy.go:24]: @Kubuxu Looks like this is where part of the problem lies.

Comment (frrist) [reply metrics/proxy.go:24]: Best guess is that something may be amiss here: https://github.com/filecoin-project/go-jsonrpc/blob/master/client.go#L284-L305

**Comment by frrist** [comment]
closing, but may refer to later as an example.

**Pull request closed without merging**